Tools/Cli-LoRa-Device-Provisioning/IoTDeviceHelper.cs

```diff
@@ -0,0 +1,1113 @@
+            {
+                StatusConsole.WriteLine(MessageType.Error, "DevEui is missing.");
+                isValid = false;
+            }
+            else if (!ValidationHelper.ValidateHexStringTwinProperty(opts.DevEui, 8, out validationError))
+            {
+                StatusConsole.WriteLine(MessageType.Error, $"DevEui {opts.DevEui} is invalid: {validationError}.");
+                isValid = false;
+            }
+            else
+            {
+                StatusConsole.WriteLine(MessageType.Info, $"DevEui {opts.DevEui} is valid.");
+            }
+
+            // ******************************************************************************************
+            // ABP device specific properties
+            // ******************************************************************************************
+            if (string.Equals(opts.Type, "ABP", StringComparison.InvariantCultureIgnoreCase))
+            {
+                // NwkSKey
+                if (string.IsNullOrEmpty(opts.NwkSKey))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, "NwkSKey is missing.");
+                    isValid = false;
+                }
+                else if (!ValidationHelper.ValidateHexStringTwinProperty(opts.NwkSKey, 16, out validationError))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, $"NwkSKey {opts.NwkSKey} is invalid: {validationError}.");
+                    isValid = false;
+                }
+                else
+                {
+                    StatusConsole.WriteLine(MessageType.Info, $"NwkSKey {opts.NwkSKey} is valid.");
+                }
+
+                // AppSKey
+                if (string.IsNullOrEmpty(opts.AppSKey))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, "AppSKey is missing.");
+                    isValid = false;
+                }
+                else if (!ValidationHelper.ValidateHexStringTwinProperty(opts.AppSKey, 16, out validationError))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, $"AppSKey {opts.AppSKey} is invalid: {validationError}.");
+                    isValid = false;
+                }
+                else
+                {
+                    StatusConsole.WriteLine(MessageType.Info, $"AppSKey {opts.AppSKey} is valid.");
+                }
+
+                // NetId
+                if (!string.IsNullOrEmpty(opts.NetId))
+                {
+                    if (!ValidationHelper.ValidateHexStringTwinProperty(opts.NetId, 3, out validationError))
+                    {
+                        StatusConsole.WriteLine(MessageType.Error, $"NetId {opts.NetId} is invalid: {validationError}.");
+                        isValid = false;
+                    }
+                    else
+                    {
+                        StatusConsole.WriteLine(MessageType.Info, $"NetId {opts.NetId} is valid.");
```

Comment:
also here, I don't think the valid is required as a message?

Reply:
I don't understand this one. Let's discuss.

Reply:
Discussed.